src/Concordium/Client/Runner.hs

```diff
@@ -3744,3 +3982,37 @@ signEncodedTransaction encPayload sender energy nonce expiry accKeys = Types.Nor
       }
       keys = Map.toList $ fmap Map.toList accKeys
   in Types.signTransaction keys header encPayload
+
+-- |Extract the response value of a GRPCResult and return it under the
+-- provided mapping.
+-- Returns @Left@ wrapping an error string describing its nature if the
+-- result contains an error, or a @Right@ wrapping the response value
+-- under the provided mapping otherwise.
+extractResponseValue :: (a -> b) -> GRPCResult (FromProtoResult a) -> Either String b
+extractResponseValue f res =
+  case res of
+    Left err -> Left $ "A GRPC error occurred: " <> err
+    Right resp ->
+      case grpcResponseVal resp of
+        Left err -> Left $ "Unable to convert response payload: " <> err
+        Right val -> Right $ f val
+
+-- |Extract the response value of a GRPCResult and return it under the
+-- provided mapping or fail with an error message if the result contains
+-- an error. Takes a string to be prepended to the error message.
+extractResponseValueOrFail :: (MonadIO m)
+  => (a -> b) -- |Result mapping
+  -> String -- |A prefix to the error message to print case of an error.
+  -> GRPCResult (Either String a) -> m b
+extractResponseValueOrFail f errPrefix res =
+  case extractResponseValue f res of
+    Left err -> logFatal [errPrefix <> err]
+    Right v -> return v
+
+-- |Get the response value of a GRPCResult or fail if the result contains an error.
```

Comment:
I will rename this.

Reply:
Please do before merging :)

Reply:
Yes. I changed this in a later PR which I will request review for today. I hope that is alright.